gtsfm/frontend/retriever/retriever_base.py

```diff
@@ -0,0 +1,64 @@
+"""Image retriever for front-end.
+Authors: Travis Driver, Jon Womack
+"""
+import abc
+from typing import List, Optional, Tuple, Dict
+import itertools
+
+import numpy as np
+import cbir
+import dask
+from dask.delayed import Delayed
+from gtsam import Pose3, Cal3Bundler
+import networkx as nx
+import pickle
+
+import gtsfm.utils.logger as logger_utils
+from gtsfm.common.keypoints import Keypoints
+
+
+logger = logger_utils.get_logger()
+
+
+class RetrieverBase(metaclass=abc.ABCMeta):
+    """Base class for image retrieval.
+    The Retriever proposes image pairs to conduct local feature matching.
+    """
+
+    def __init__(self, image_pair_indices):
+        """Initialize the Retriever
+
+        :param image_pair_indices: Set of possible image pairs based on Dataloader
+        """
+        self.image_pair_indices = image_pair_indices
+        orb_descriptor = cbir.descriptors.Orb()
```

Comment:
let's add cbir to the conda env?

Reply:
I'm installing cbir with this [repo](https://github.com/epignatelli/scalable-recognition-with-a-vocabulary-tree/tree/master/cbir). Should I add it to thirdparty and update the Readme with installation `pip install -e /thirdparty/scalable-recognition-with-a-vocabulary-tree/cbir/`?